posthog/api/feature_flag.py

```diff
@@ -121,9 +129,121 @@ def get_queryset(self) -> QuerySet:
         return queryset.order_by("-created_at")
 
     @action(methods=["GET"], detail=False)
-    def for_user(self, request: request.Request, **kwargs):
-        distinct_id = request.GET.get("distinct_id", None)
-        if not distinct_id:
-            raise serializers.ValidationError("Please provide a distinct_id to continue.")
-        flags = get_active_feature_flags(self.team, distinct_id)
-        return Response({"distinct_id": distinct_id, "flags_enabled": flags})
+    def my_flags(self, request: request.Request, **kwargs):
+        if not request.user.is_authenticated:  # for mypy
+            raise exceptions.NotAuthenticated()
+        feature_flags = (
+            FeatureFlag.objects.filter(team=self.team, active=True, deleted=False)
```

Comment:
At this point it might be okay to keep this here, but for future reference I think this logic belongs in the serializer. The serializer constructs and handles any response parsing/encoding while the view just processes the request for the serializer.

Reply:
Got it - thanks! I've definitely been a bit confused about what exactly goes in the serializer vs view. This is really helpful.

I want to get this PR in, so it doesn't linger more, but I'll put up another PR that cleans this up after.